gno.land/pkg/integration/testing_integration.go

```diff
@@ -0,0 +1,257 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"hash/crc32"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/node"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys/client"
+	"github.com/gnolang/gno/tm2/pkg/log"
+	"github.com/jaekwon/testify/require"
+	"github.com/rogpeppe/go-internal/testscript"
+)
+
+// XXX: This should be centralize somewhere.
+const (
+	test1Addr = "g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5"
+	test1Seed = "source bonus chronic canvas draft south burst lottery vacant surface solve popular case indicate oppose farm nothing bullet exhibit title speed wink action roast"
+)
+
+type testNode struct {
+	*node.Node
+	logger      log.Logger
+	nGnoKeyExec uint // Counter for execution of gnokey.
+}
+
+func SetupGnolandTestScript(t *testing.T, txtarDir string) testscript.Params {
+	t.Helper()
+
+	cmd := exec.Command("go", "list", "-m", "-mod=mod", "-f", "{{.Dir}}", "github.com/gnolang/gno")
+	out, err := cmd.CombinedOutput()
+	require.NoError(t, err)
+
+	// `gnoRootDir` should point to the local location of the gno repository.
+	// It serves as the gno equivalent of GOROOT.
+	gnoRootDir := strings.TrimSpace(string(out))
+
+	// `gnoHomeDir` should be the local directory where gnokey stores keys.
+	gnoHomeDir := filepath.Join(t.TempDir(), "gno")
+
+	// `gnoDataDir` should refer to the local location where the gnoland node
+	// stores its configuration and data.
+	gnoDataDir := filepath.Join(t.TempDir(), "data")
+
+	var muNodes sync.Mutex
+	nodes := map[string]*testNode{}
+
+	persistWorkDir, _ := strconv.ParseBool(os.Getenv("TESTWORK"))
```

Comment:
(good to document it in the `docs/` file I mentioned)

OR we can remove this, in favour of LOG_DIR

Reply:
`TESTWORK` is not specifically intended for debugging logs; however, it is useful when you want to make the txtar's current working directory persistent for debugging. It also provides another reason why I should document this more thoroughly 🙂.